WordPress/gutenberg
# [RNMobile] Move some Unsupported Block Editor injection code to gutenberg-mobile repo

gb-mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2561
WPiOS side PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/14683
WPAndroid side PR: https://github.com/wordpress-mobile/WordPress-Android/pull/12795

## Description
This PR removes some references to `window.wp.data.dispatch( 'automattic/nux' );` and ` "automattic/nux":` and implements a mechanism to insert custom JS to be injected from the client.

To achieve this, some changes were needed on when the JS code is injected:

- There's a new function `gutenbergObserver` which will check when gutenberg is ready for the post content to be inserted. It will send a `gutenbergReady` message to the native side.
  - This removes the need to check for requests to `"/wp-admin/post-new.php"` in order to insert the content, making the whole system more robust.

- When the `gutenbergReady` call is received we:
  - Run the "external JS sources" (from WP Mobile Apps).
  - Run the autosave prevention script.
  - Run the insert block script.

## How has this been tested?




To test (On both WPiOS and WPAndroid apps):

- Create a new wpcom account with a new simple site.
- On the newly created site, create a new gutenberg post.
- On HTML mode, paste this snippet (or any other unsupported block HTML code): ``
- Go back to visual mode.
- Open the unsupported block with the Unsupported Block Editor.
- Check that Gutenberg editor loads without the NUX UI covering it.


## Checklist:
